feat/post

chaeppy opened the pull request.

chaeppy committed:
[ADD] 회원가입, 로그인, 로그아웃

chaeppy committed:
[ADD] ExceptionResponse

hyeinisfree committed:
update: .gitignore

chaeppy committed:
[update] Account Entity 보완

chaeppy committed:
[add] Post Entity

chaeppy committed:
[add] PostLike Entity

chaeppy committed:
[add] PostImages Entity

chaeppy committed:
[add] PostComment Entity

chaeppy committed:
[update] jpa 설정 보완
스키마 정보 일치 체크
네이밍 전략 명시화

chaeppy committed:
Merge branch 'main' of https://github.com/SSWU-Community/spring-study into post

chaeppy committed:
[ADD] Post 모듈 추가

chaeppy committed:
[ADD] Account Service getUserInfo()에 readOnly 속성 추가

chaeppy committed:
[ADD] NotFountException

chaeppy committed:
[ADD] PostCreateRequestDto에 entity로 변경 함수 추가

chaeppy committed:
[ADD] PostRepository에 findAllByAccountOrderByCreatedAt 함수 추가

chaeppy committed:
[ADD] PostResponseDto

chaeppy committed:
[ADD] PostService에 findAllByAccount,findAll,findById,save,delete 험수 추거

chaeppy committed:
[Comment] AuthenticationPrincipal에 대한 설명

chaeppy committed:
[Refactor] 함수 이름 더 직관적으로 수정

chaeppy committed:
[Add] findAll,findById,save,delete 기능 구현

chaeppy committed:
[Refactor] 코드 포맷팅 , 안쓰는 import문 삭제

chaeppy committed:
[Fix] 빌더 사용시 필드 초기화 제대로 되지 않던 문제 해결

chaeppy committed:
[Fix] 게시글 최신순 정렬 해결

chaeppy committed:
[Feat] 생성시간, 수정시간 자동 설정해주는 BaseEntity추가

chaeppy committed:
[Feat] BaseEntity사용위해 EnableJpaAuditing 추가

chaeppy committed:
[Feat] PostEntity는 BaseEntity 상속받고 기존 CreatedAt 필드 삭제

chaeppy committed:
[Test] PostRepositoryTest 추가

chaeppy committed:
[Chore] 테스트시 h2 DB 이용 위해 추가

chaeppy committed:
[Chore] 테스트용 application.yml 추가
h2사용

chaeppy committed:
[Feat] Post 객체 비교 위해 @EqualsAndHashCode 추가

chaeppy committed:
[Feat] Test에서 Dto객체 사용 위해 Builder어노테이션 추가

chaeppy committed:
[Feat] findAll, save 함수 추가

chaeppy committed:
[Test] findAllByAccount,findById,delete 테스트 추가

chaeppy committed:
[Refactor] 공통부분 findByIdOrThrowNotFoundException 으로 함수화

chaeppy committed:
[Feat] 게시글 생성,수정 PostRequestDto로 공통화. 수정을 위한 apply함수 추가

chaeppy committed:
[Feat] PostController에 update추가. PostRequestDto로 공통화.

chaeppy committed:
[Test] PostRequestDto로 공통화로 인한 수정

chaeppy committed:
[Feat] PostService에 Update함수 추가, save함수 응답값 수정, PostRequestDto 공통화

chaeppy committed:
[Feat] Post Entity에 update함수 추가. account 지연로딩 삭제
account 지연로딩 삭제이유: 프록시 관련 오류발생

hyeinisfree commented on src/main/java/sungshin/sooon/service/PostService.java:
```diff
@@ -0,0 +1,96 @@
+        Post post = postRequestDto.toPost();
+        post.setAccount(account);
+        return PostResponseDto.of(postRepository.save(post));
+    }
```
account가 null일때는 어떻게 되나요?

chaeppy replied:
@hyeinisfree  Controller부터 타고 들어온다고 생각하면 Account가 Authentication되어서 SecurityContext에 저장된애를 받아온거라 null인경우가 절대 없을텐데 (인증성공되지 않으면 error 응답) 
Service Layer만 보면 Account가 null인 경우를 체크해야하나? 싶긴 하네요

hyeinisfree replied:
음 그렇네요 인증 오류가 나니까

hyeinisfree commented on src/main/java/sungshin/sooon/service/PostService.java:
```diff
@@ -0,0 +1,96 @@
+        postRepository.delete(post);
+    }
+
+    private Post findByIdOrThrowNotFoundException(Long id){
```
이거 왜 따로 함수로 만드셨나요? postRepositoty.findById(id).orElseThrow() 어쩌구 있지 않나요?

sio2whocodes replied:
따로 만들고 싶으셨나부져

chaeppy replied:
@hyeinisfree 

findById,delete 함수 에서 중복적으로 쓰이는 코드라 함수로 뺐습니다요
그리고 findById에선 리턴값이 Dto입니다

hyeinisfree replied:
@chaeppy 

findById에서 리턴값이 왜 Dto 인가요?

chaeppy replied:
@hyeinisfree 
service 의 findById 함수요 !!

hyeinisfree commented on src/main/java/sungshin/sooon/domain/entity/Post.java:
헉 이거 이렇게 해야 되나요?

chaeppy replied:
@hyeinisfree 양쪽 다 연관관계 추가해야하는거 말씀하시는거면 그렇습니다.. (양방향 매핑의 경우)
단방향 매핑이면 연관관계의 주인(n:1에서 n인 테이블, fk를 갖고 있는 테이블)에만 설정해주면 되구요.
양방향 매핑이면 서비스단에서 두개다 이렇게 설정해주거나 제 코드처럼 연관관계편의메소드를 사용해주시면 됩니다.
https://leejaedoo.github.io/relationship_mapping/

chaeppy commented on src/main/java/sungshin/sooon/controller/PostController.java:
```diff
@@ -0,0 +1,78 @@
+        return new ResponseEntity(posts,HttpStatus.OK);
+    }*/
+
+    @GetMapping("/")
```
```suggestion
    @GetMapping("")
```

https://sanghaklee.tistory.com/57 

> 후행 슬래쉬(/)는 URI에 포함되지 않아야 합니다. 
계층관계를 나타낼 때 슬래시 구분자를 사용해야합니다.
출처: https://sas-study.tistory.com/265

chaeppy commented on src/main/java/sungshin/sooon/controller/PostController.java:
```diff
@@ -0,0 +1,78 @@
+        return new ResponseEntity(post, HttpStatus.OK);
+    }
+
+    @PostMapping("/")
```
```suggestion
    @PostMapping("")
```

chaeppy commented on src/main/java/sungshin/sooon/domain/entity/PostComment.java:
```diff
@@ -0,0 +1,39 @@
+    @JoinColumn(name = "post_id")
+    private Post post;
+
+    @ManyToOne(fetch = FetchType.LAZY) //주로 익명이므로
```
Lazy로 하면 Post-Account 사례쳐럼 오류날 것 같다.
지연로딩시 문제점 조사필요

chaeppy committed:
Style: post 후행 슬래쉬 삭제

chaeppy committed:
Style: get후행 슬래쉬 삭제

chaeppy committed:
Fix: Entity 초기화 문제 해결

chaeppy committed:
Remove: application.yml

chaeppy committed:
Docs: gitignore수정

chaeppy committed:
Fix: 비밀번호 밸리데이션 정규식 잘못되있던 것

chaeppy committed:
Style: PostController pathVariable 변수명 변경

chaeppy committed:
Add: 추가 like,unlike service

chaeppy committed:
Add: 삭제 PoskLike setter

chaeppy committed:
Add: 추가 like, unlike api

chaeppy committed:
Feat: 수정 회원가입,로그인 api 응답시 유저정보와 토큰반환

chaeppy committed:
Style: 수정 이메일 중복체크 오류시 응답 메세지

chaeppy committed:
Feat: 추가 PostLikeRepository

chaeppy committed:
Feat: 추가 좋아요 여부 조회 api, 수정 좋아요/좋아요 취소 uri

chaeppy committed:
Feat: 추가 좋아요 여부 조회 service, 추가 좋아요 취소 service 권한체크

chaeppy committed:
Feat: 추가 post 조회 응답에 좋아요 개수

chaeppy committed:
Feat: 추가 post entity 생성시 likeCount 서브쿼리

chaeppy committed:
임시저장

chaeppy committed:
Fix: Dto의 isAnonymous가 제대로 validation 되지 않던 문제 해결

정확한 사유는 아니지만 해당 문구를 보고 is를 제거하였더니 해결됨
validation Annotation을 field에 적용할 수 있지만 getter에도 적용할 수 있습니다. boolean의 경우 is나 has로 시작하는 경우에 해당합니다.
DefaultGetterPropertySelectionStrategy 클래스를 보면 “get”, “is”, “has”로 시작하는 메서드인지 확인하는 코드가 있습니다.
https://kapentaz.github.io/java/Java-Bean-Validation-%EC%A0%9C%EB%8C%80%EB%A1%9C-%EC%95%8C%EA%B3%A0-%EC%93%B0%EC%9E%90/#

chaeppy committed:
Fix: 익명여부 CommentRequestDto에 들어오지 않던 문제 해결

chaeppy committed:
Feat: 수정 CommentResponseDto 응답값

chaeppy committed:
Style: 코드 포맷팅, 안쓰는 패키지 import 제거

chaeppy committed:
Feat: 제거 @Setter, 추가 setOrderNum 함수

chaeppy committed:
Feat: 추가 PostComment Api

chaeppy committed:
Feat: 추가 Post 전체 조회 Paging 기능

chaeppy committed:
Style: 변경 좋아요 개수 변수명

chaeppy committed:
Chore: 추가 S3관련 설정

chaeppy committed:
Feat: 추가 S3 이미지 업로드 유틸

chaeppy committed:
Feat: 추가 이미지 업로드 컨트롤러
